lib/cron.dart

```diff
@@ -201,67 +226,67 @@ class _Cron implements Cron {
     _timer = null;
     final now = clock.now();
     for (final schedule in _schedules) {
-      schedule.tick(now);
+      final job = schedule.tick(now);
+      if (job != null) {
+        _jobManager.start(job, schedule.task);
+      }
     }
     _scheduleNextTick();
   }
 }
 
 class _ScheduledTask implements ScheduledTask {
+
   @override
-  final Schedule schedule;
-  final Task _task;
+  String get id => '${task.hashCode}';
 
-  bool _closed = false;
-  Future? _running;
-  bool _overrun = false;
+  @override
+  final Schedule schedule;
 
   @override
-  bool get isRunning => _running != null;
+  final Task task;
+
+  bool _closed = false;
 
   /// The datetime a Task last run.
   DateTime lastTime = DateTime(0, 0, 0, 0, 0, 0, 0);
 
-  _ScheduledTask(this.schedule, this._task);
+  _ScheduledTask(this.schedule, this.task);
 
-  void tick(DateTime now) {
-    if (_closed) return;
-    if (!schedule.shouldRunAt(now)) return;
+  Job? tick(DateTime now) {
+    if (_closed) return null;
+    if (!schedule.shouldRunAt(now)) return null;
     if ((schedule.seconds == null || lastTime.second == now.second) &&
         (schedule.minutes == null || lastTime.minute == now.minute) &&
         (schedule.hours == null || lastTime.hour == now.hour) &&
         (schedule.days == null || lastTime.day == now.day) &&
         (schedule.months == null || lastTime.month == now.month) &&
         (schedule.weekdays == null || lastTime.weekday == now.weekday)) {
-      return;
+      return null;
     }
     lastTime = now;
-    _run();
+    return Job(taskId: id, id: '$id-${now.millisecondsSinceEpoch}');
   }
 
-  void _run() {
-    if (_closed) return;
-    if (_running != null) {
-      _overrun = true;
-      return;
-    }
-    _running =
-        Future.microtask(() => _task()).then((_) => null, onError: (_) => null);
-    _running!.whenComplete(() {
-      _running = null;
-      if (_overrun) {
-        _overrun = false;
-        _run();
-      }
-    });
-  }
+  // void _run() {
+  //   if (_closed) return;
+  //   if (_running != null) {
+  //     _overrun = true;
+  //     return;
+  //   }
+  //   _running =
+  //       Future.microtask(() => _task()).then((_) => null, onError: (_) => null);
+  //   _running!.whenComplete(() {
+  //     _running = null;
+  //     if (_overrun) {
+  //       _overrun = false;
+  //       _run();
+  //     }
+  //   });
+  // }
 
   @override
   Future<void> cancel() async {
     _closed = true;
-    _overrun = false;
```

Comment:
The _overrun is not used anymore since the ScheduledTask is just a template for the Job

Reply:
I'm more interested in the why: why is not needed to track it anymore? Related: what if the scheduler wants the next execution to wait until the previous completes its running? Shouldn't that be a configuration option and this way we don't break the runtime behavior for existing clients?

Reply:
I thought it wasn't needed anymore since in the proposed implementation there is not a way to wait for a job to finish. But you are totally right this is a breaking change and should be added an option to manage the jobs like before. We can, however, prevent the breaking change making the option enabled by default.